lexicon/atrium-codegen/src/token_stream.rs

```diff
@@ -562,13 +562,26 @@ fn string_type(string: &LexString) -> Result<(TokenStream, TokenStream)> {
     Ok((description, typ))
 }
 
-fn unknown_type(unknown: &LexUnknown, name: Option<&str>) -> Result<(TokenStream, TokenStream)> {
+enum NamedUnknown<'s> {
+    Field(&'s str),
+    Array(&'s str),
+}
+
+fn unknown_type(
+    unknown: &LexUnknown,
+    name: NamedUnknown<'_>,
+) -> Result<(TokenStream, TokenStream)> {
     let description = description(&unknown.description);
-    if name == Some("didDoc") {
-        Ok((description, quote!(crate::did_doc::DidDocument)))
-    } else {
-        Ok((description, quote!(crate::records::Record)))
-    }
+
+    let typ = match name {
+        NamedUnknown::Field("didDoc") => quote!(crate::did_doc::DidDocument),
+        NamedUnknown::Field("record") | NamedUnknown::Array("relatedRecords") => {
+            quote!(crate::records::Record)
```

Comment:
I figured that `unknown` fields named `record` or `relatedRecords` should still use `Record`, so after changing the default to `Unknown` I added these two extra cases, which affected the following lexicons: https://github.com/sugyan/atrium/compare/2b1747594b9e1fd711662416971c28c3cd7e1843..e1d4b5ca1e4c3ab9e891086bd4e2200c549ff742